uxarray/helpers.py

```diff
@@ -8,14 +8,14 @@
 config.DISABLE_JIT = False
 
 
-def parse_grid_type(filepath, **kw):
+def parse_grid_type(dataset, **kw):
     """Checks input and contents to determine grid type. Supports detection of
     UGrid, SCRIP, Exodus and shape file.
 
     Parameters
     ----------
-    filepath : str
-       Filepath of the file for which the filetype is to be determined.
+    dataset : Xarray dataset
+       Xarray dataset of the file input
```

Comment:
The term "file input" might be confusing. As we are not getting a file inut anymore, let's say "... of the grid."